numpy/polynomial/polynomial.py

```diff
@@ -1441,27 +1441,31 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None):
 
     Examples
     --------
-    >>> np.random.seed(123)
     >>> from numpy.polynomial import polynomial as P
     >>> x = np.linspace(-1,1,51)  # x "data": [-1, -0.96, ..., 0.96, 1]
-    >>> y = x**3 - x + np.random.randn(len(x))  # x^3 - x + Gaussian noise
+    >>> rng = np.random.default_rng(seed=123)
+    >>> err = rng.standard_normal(size=len(x))
+    >>> y = x**3 - x + err  # x^3 - x + Gaussian noise
     >>> c, stats = P.polyfit(x,y,3,full=True)
-    >>> np.random.seed(123)
-    >>> c # c[0], c[2] should be approx. 0, c[1] approx. -1, c[3] approx. 1
-    array([ 0.01909725, -1.30598256, -0.00577963,  1.02644286])  # may vary
+    >>> c # c[0], c[1] approx. -1, c[2] should be approx. 0, c[3] approx. 1
+    array([ 0.23111996, -1.02785049, -0.2241444 ,  1.08405657]) # may vary
     >>> stats # note the large SSR, explaining the rather poor results
-     [array([ 38.06116253]), 4, array([ 1.38446749,  1.32119158,  0.50443316, # may vary
-              0.28853036]), 1.1324274851176597e-014]
+    [array([48.312088]),                                      # may vary
+     4,                                                       # may vary
+     array([1.38446749, 1.32119158, 0.50443316, 0.28853036]), # may vary
+     1.1324274851176597e-14]                                  # may vary
 
     Same thing without the added noise
 
     >>> y = x**3 - x
     >>> c, stats = P.polyfit(x,y,3,full=True)
-    >>> c # c[0], c[2] should be "very close to 0", c[1] ~= -1, c[3] ~= 1
-    array([-6.36925336e-18, -1.00000000e+00, -4.08053781e-16,  1.00000000e+00])
+    >>> c # c[0], c[1] ~= -1, c[2] should be "very close to 0", c[3] ~= 1
+    array([-6.73496154e-17, -1.00000000e+00,  0.00000000e+00,  1.00000000e+00])
     >>> stats # note the minuscule SSR
-    [array([  7.46346754e-31]), 4, array([ 1.38446749,  1.32119158, # may vary
-               0.50443316,  0.28853036]), 1.1324274851176597e-014]
+    [array([8.79579319e-31]),                                 # may vary
+     4,                                                       # may vary
+     array([1.38446749, 1.32119158, 0.50443316, 0.28853036]), # may vary
+     1.1324274851176597e-14]                                  # may vary
```

Comment:
I would love some feedback on this particular change. Rather than try to cut a really long output at one spot, I used the Jupyter formatting.  Good/bad?

Reply:
As long as doctests are OK with it, I think this is better.

Reply:
Looks like the tests passed.  

Though something is off with the docstring tester, as I purposefully entered some arbitrary changes to make it fail, yet the tests passed. I'll be exploring that more and open an issue if/when I have something reproducible with a bit more to go on.